### ISSUE TITLE
Fatal error: video info not available

### ISSUE THREAD

**Issue by lhc70000** [iina/iina#3013]
`PlayerWindowController L427`

A user confirmed that this could happen for some live streams.

We should consider continuing the playback even if `videoDuration` and/or `videoPosition` cannot be obtained.

**Comment by matellis** [comment]
I am getting this all the time when using a NAS. Why does it have to be a fatal error, couldn't it just refuse to open the file?

**Comment by matellis** [comment]
Submitted pull request with fix

**Comment by low-batt** [comment]
This issue has been reported again in #3266.

I have hit this failure multiple times. I am able to reproduce it on my 2019 MacBook Air running macOS 10.15.7. I use a bash script to start 20 separate instances of IINA all at once. Usually somewhere between 1-3 fail. Rarely do all 20 work.

The failure is a result of a race condition between threads when mpv is opening the file. Adding some additional logging before the call to `fatal` in `updatePlayTime` shows mpv has not finished loading the file and position and duration are unknown, tripping the fatal internal error check:

>18:10:20.721 [iina][d] <NSThread: 0x7fc922605d50>{number = 1, name = main} PlayerWindowController.updatePlayTime(withDuration:andProgressBar:):438 14.mp4 player.info.justStartedFile false  **player.info.fileLoading true** player.info.videoPosition:  nil player.info.videoDuration:  nil 

While mpv is in the process of opening the file it is emitting events for properties that have been updated. The function `MPVController.readEvents` is being called to process the events. That function dispatches a call to `MPVController.handleEvent` in a background thread. The `handleEvent` function calls `handlePropertyChange`. One of the properties mpv updates is `MPVProperty.chapter`. When that property is updated `handlePropertyChange` calls `player.syncUI`. Still running in that background thread the function `syncUI` stores the current values of the mpv properties `MPVProperty.timePos` and `MPVProperty.duration` in `player.info` and then queues an asynchronous call to `PlayerWindowController.updatePlayTime`. As mpv is still loading the file while the event is being processed by the background thread the position and duration properties may or may not have been set when `syncUI` captures the values of these properties. If either of the values stored in `player.info` are still `nil` when `updatePlayTime` is run then the fatal error will be triggered.

The change in the pull request updates the first `guard` statement in `updatePlayTime` to not process the update if either position or duration are unknown and removes the second `guard` statement containing the fatal error:
```swift
// IINA listens for changes to mpv properties such as chapter that can occur during file loading
// resulting in this function being called before mpv has set its position and duration
// properties. If position or duration are unknown assume the file is still being loaded.
guard loaded, let duration = player.info.videoDuration, let pos = player.info.videoPosition else { return }
```

The OSD also displays position and duration. The code in `MPVController.handleEvent` that forms a string representation of the position and duration for the OSD uses placeholders if the data is not available.  Something similar could be done in `updatePlayTime` but that would be a much larger code change.

Thoughts?

**Comment by low-batt** [comment]
The fix in PR #3434 has been merged into the develop branch.

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.
